relay-metrics/src/aggregation.rs

```diff
@@ -745,6 +745,46 @@ impl BucketKey {
         std::hash::Hash::hash(self, &mut hasher);
         hasher.finalize() as i64
     }
+
+    /// Validates the metric name and tags' keys and values fulfill the max length threshold.
+    ///
+    /// If the metric name surpasses the threshold, returns an Err. Tags with surpassing key or
+    /// value lengths are removed from the bucket.
+    fn validate_string_lengths(mut self, aggregator_config: &AggregatorConfig) -> Result<Self, ()> {
+        if self.metric_name.len() as u64 > aggregator_config.max_name_length {
+            relay_log::error!(
+                "Invalid metric name, too long ({} > max:{}): {:?}",
+                self.metric_name.len(),
+                aggregator_config.max_name_length,
+                self.metric_name
+            );
+            return Err(());
+        }
+
+        self.tags.retain(|key, value| {
+            if key.len() as u64 > aggregator_config.max_name_length {
+                relay_log::error!(
+                    "Invalid tag key, too long ({} > max:{}): {:?}",
+                    key.len(),
+                    aggregator_config.max_name_length,
+                    key
+                );
+                return false;
+            }
+            if value.len() as u64 > aggregator_config.max_name_length {
```

Comment:
Let's have separate configs for at least metric names and tags, ideally 3 configs for name, key, value.